jonathanhyperion/ChallengeMobileApp
# Login: Authentication, UI/UX, Navigation, Unit Test Integration.

**What's new?**
This pull request encompasses a series of significant enhancements and additions to the codebase:

- Implemented a visually appealing and user-friendly login user interface (UI) ensuring an intuitive login experience for end-users.

- Established a secure and efficient connection between the frontend and backend for `login` functionality. This integration ensures seamless communication, allowing users to authenticate securely and access the application's features seamlessly.

- Upgraded the application's navigation capabilities by integrating the [FlowStacks](https://github.com/johnpatrickmorgan/FlowStacks) dependency. This integration empowers the application with robust navigation features, enabling seamless navigation between different screens and enhancing overall user engagement.

- Implemented a unit test to validate the handling of successful login responses within the `AuthApiService`. This ensures that the authentication service effectively processes and responds to login attempts.

**How should this be tested? / Does QA need to test this?**
Yes, QA is required to conduct thorough testing on the following aspects post-merge:
1. Validate the unit test for successful login responses in `AuthApiService` to ensure accurate handling of API responses

2. Test session persistence functionality to confirm consistent user sessions across the application.

3. Thoroughly evaluate the login UI design for both aesthetics and usability, ensuring an intuitive and visually pleasing user experience.

**Do these changes have associated tests?**
Yes.

**Attach screenshots/videos to this PR**

https://github.com/jonathanhyperion/ChallengeMobileApp/assets/129110575/d5ab1e8e-fc64-419f-8bef-e6c7100894d6
